### DISCUSSION TITLE
Welcome to EMBArk Discussions!

### DISCUSSION THREAD

**Discussion by m-1-k-3** [e-m-b-a/embark#52] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by torabi12** [comment]
Hello All,

I am a software tester working on the operation of industrial solutions. These are usually complex products where firmware is a key focus in terms of operation and security.
The e-m-b-a is in my experience, the most comprehensive test tool that can be used extensively and is very efficient if it can use enough RAM and CPUs. :)

I would like to use the examples of AttifyOS and EmbedOS to put together a new Ubuntu 22.04 operating system, that includes the tools that are essential for firmware testing.
EMBA, FACT_core, firmwalker, firmware-mod-kit, firmware-analysis-toolkit, firmadyne, checksec.sh, Ghidra and other tools would be on my list.

The difficulty is that these tools are sometimes problematic to install on their own, but I hope to be able to package them into a common system.

Good luck and perseverance to everyone with the firmware analysis!

**Reply by m-1-k-3** [reply]
Hi @torabi12,

Thank you for your great feedback: 

_The e-m-b-a is in my experience, the most comprehensive test tool that can be used extensively and is very efficient if it can use enough RAM and CPUs. :)_

Your testing of EMBA and EMBArk and your feedback is really great and very helpful. Thank you very much for your support! Keep on going :)

Best,
Mike

**Comment by jwierzch** [comment]
Hey There,

I am not sure what the best way to reach out is. I want to contribute to the project. I am currently working in a dev installed and have added microsoft oauth id  sso on top. Not sure what best way to format a pull request for it (not quite ready need a little bit of cleanup)
but since i am not very familiar with django projects I wanted to maybe link up about my changes and see if there is a better way . different way to fit it into the project (where to decide how the sso code is incorporated - switches  more work up front to get switches in all the code or different install version, easier now to rip our the non sso user views etc, harder to maintain separate versions)

**Reply by BenediktMKuehne** [reply]
Hey,
that sounds exciting, I've been changing some things, relating to the user/login in the latest PR.
Adding oauth would be a great next step.
Anything that works with the default _AbstractUser_ model.

If you have a draft, I'd be happy to look at it

**Reply by jwierzch** [reply]
well tbh im not familiar with the user models in django. i was going to try and go the https://django-auth-adfs.readthedocs.io/en/latest/index.html, implemented as such: https://github.com/jwierzch/django_auth_example
i do have another poc implementation of azure sso working on the dev embark server but i dont think its nearly as clean as the above. Ill get that done in the next few nights now that i know how to use the django-auth-adfs
